v2/result.go

```diff
@@ -26,3 +26,22 @@ type HealthResult struct {
 	Ok            bool          `json:"ok"`
 	Severity      uint8         `json:"severity,omitempty"`
 }
+
+func ComputeOverallStatus(result *HealthResult) bool {
```

Comment:
Why not make the `HealthResult` a receiver and have the method automatically set `result.Ok` ?